recipe/deploy/vendors.php

```diff
@@ -27,11 +32,19 @@
     }
 
     $composerVersionToInstall = get('composer_version', null);
+    $composerChannel = get('composer_channel', null);
     $installCommand = "cd {{release_path}} && curl -sS https://getcomposer.org/installer | {{bin/php}}";
 
     if ($composerVersionToInstall) {
         $installCommand .= " -- --version=" . $composerVersionToInstall;
     }
+    elseif ($composerChannel) {
+        $composerValidChannels = ['stable', 'snapshot', 'preview', '1', '2', ];
+        if(!in_array($composerChannel, $composerValidChannels)) {
+            throw new \Exception('Selected Composer channel '.$composerChannel.' is not valid. Valid channels are: '.implode(', ', $composerValidChannels));
+        }
+        $installCommand .= " -- --".$composerChannel;
```

Comment:
Does it need additional `--`?

Reply:
Yes it does. Usage is:
```shell
$ php composer-installer.php -- --1
$ php composer-installer.php -- --2
$ php composer-installer.php -- --snapshot
$ php composer-installer.php -- --preview
```

Reply:
Will it work with a specified composer version? See a few lines above.

Reply:
I don't think it will, which is why I've used an `elseif`. I'm not sure it would even make sense: if you want a specific version, use `composer_version`, if you want the latest version of a channel, use `composer_channel`?

Reply:
Do it. Thanks.